File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
from itertools import product

def solution(word):
    alphabets = ['A','E','I','O', 'U']
    
    dictionary = []
    
    for i in range(1,6):
        #같은 알파벳 반복해서 넣기, 1일 땐 순열로 만듦
        #if i != 1:
        #    for a in alphabets:
        #        dictionary.append(a*i)
            
        #순열 만들기
        for p in product(alphabets, repeat = i):
            w = ''.join(p).strip()
            dictionary.append(w)
    
    dictionary.sort()
    #print(dictionary)
    answer = dictionary.index(word)+1
    
    return answer

File: 프로그래머스/2/84512. 모음 사전/모음 사전.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string word) {
    int answer = 0;
    int weight[5] = {0,};
    
    int w = 1;
    weight[4] = 1;
    
    for (int i=3; i>=0; i--) {
        w = w * 5;
        weight[i] = w + weight[i+1];
    }
    
    string alpha = "AEIOU";
    
    for (int i=0; i<word.size(); i++) {
        int index = alpha.find(word[i]);
        answer += index * weight[i] + 1;
    }
    
    return answer;
}